ambry-replication/src/test/java/com.github.ambry.replication/BlobIdTransformerTest.java

```diff
@@ -297,17 +346,25 @@ private void assertException(TransformationOutput transformationOutput, Class ex
      */
     InputAndExpected(Pair<String, String> pair, Class clazz, boolean divergeInfoFromData)
         throws IOException, MessageFormatException {
+      this(pair, clazz, divergeInfoFromData, ChunkTypes.IS_DATA);
+    }
+
+    InputAndExpected(Pair<String, String> pair, Class clazz, boolean divergeInfoFromData, ChunkTypes chunkType)
+        throws IOException, MessageFormatException {
       boolean hasEncryption = clazz == PutMessageFormatInputStream.class;
       Long crcInput = buildRandom.nextLong();
-      input = buildMessage(pair.getFirst(), clazz, hasEncryption, crcInput, divergeInfoFromData);
+      input = buildMessage(pair.getFirst(), clazz, hasEncryption, crcInput, divergeInfoFromData, chunkType);
       if (pair.getSecond() == null) {
         //can't just assign 'input' since Message has an
         //InputStream that is modified when read
         expected = null;//buildMessage(pair.getFirst(), PutMessageFormatInputStream.class, hasEncryption);
       } else {
         Long crcExpected = pair.getSecond().equals(pair.getFirst()) ? crcInput : null;
+        if (chunkType == ChunkTypes.IS_METADATA) {
+          chunkType = ChunkTypes.IS_METADATA_EXPECTED;
```

Comment:
it's a bit confusing that you are overwriting an argument. Maybe clearer to declare another variable or do this inline

Reply:
removed enum